src/xrc_mock/src/lib.rs

```diff
@@ -1,135 +1,52 @@
-use ic_cdk::{println, update};
-use std::{cell::RefCell, collections::HashMap};
-
-pub const XRC_REQUEST_CYCLES_COST: u128 = 1_000_000_000;
+use ic_xrc_types::{
+    Asset, AssetClass, ExchangeRate, ExchangeRateMetadata, GetExchangeRateRequest,
+    GetExchangeRateResult,
+};
+use std::cell::RefCell;
 
 thread_local! {
-    static DATA: RefCell<HashMap<u64, (u64, u32)>> = RefCell::new(HashMap::new());
-}
-
-/// Set test data: [(time in seconds since epoch, (rate, decimals))]
-#[update]
-pub fn set_exchange_rate_data(data: Vec<(u64, (u64, u32))>) {
-    DATA.with_borrow_mut(|map| {
-        for (k, v) in data.into_iter() {
-            map.insert((k / 60) * 60, v);
-        }
-    })
-}
-
-#[update]
-pub fn get_exchange_rate(request: GetExchangeRateRequest) -> GetExchangeRateResult {
-    let payment = ic_cdk::api::msg_cycles_accept(XRC_REQUEST_CYCLES_COST);
-    if payment < XRC_REQUEST_CYCLES_COST {
-        return GetExchangeRateResult::Err(ExchangeRateError::NotEnoughCycles);
-    }
-
-    let GetExchangeRateRequest {
+    static RATES: RefCell<Vec<(u64, u64)>> = const { RefCell::new(vec![]) } // (timestamp, rate)
+}
+
+const CALL_CYCLES_COST: u128 = 1_000_000_000;
+
+#[ic_cdk::update]
+async fn get_exchange_rate(request: GetExchangeRateRequest) -> GetExchangeRateResult {
+    ic_cdk::api::msg_cycles_accept(CALL_CYCLES_COST);
+    // Find the closest rate to the request timestamp
+    let timestamp = request
+        .timestamp
+        .unwrap_or_else(|| (ic_cdk::api::time() / 1_000_000_000) + 6);
+    let rates = RATES.with(|rates| rates.borrow().clone());
+    let closest_rate = rates.iter().min_by_key(|(t, _)| t.abs_diff(timestamp));
+    let default_rate = (timestamp, 3_497_900_000); // 1 ICP = 3.4979 XDR
+    let (_, rate) = closest_rate.unwrap_or(&default_rate);
+    GetExchangeRateResult::Ok(ExchangeRate {
+        base_asset: Asset {
+            symbol: "ICP".to_string(),
+            class: AssetClass::Cryptocurrency,
+        },
+        quote_asset: Asset {
+            symbol: "CXDR".to_string(),
```

Comment:
CXDR is a non-standard thing invented by the exchange rate canister (and is not documented here).

Maybe add a small comment that says something like

> "C" stands for "computed" (and "XDR" is the standard symbol for standard drawing rights).